ibm/config.go

```diff
@@ -1299,6 +1311,32 @@ func (c *Config) ClientSession() (interface{}, error) {
 	}
 	session.appidAPI = appIDClient
 
+	// Construct an "options" struct for creating Context Based Restrictions service client.
+	cbrURL := contextbasedrestrictionsv1.DefaultServiceURL
+	if c.Visibility == "private" || c.Visibility == "public-and-private" {
+		session.contextBasedRestrictionsClientErr = fmt.Errorf("Context Based Restrictions Service API does not support private endpoints") //return this error if private endpoints are not supported
+	}
+	if fileMap != nil && c.Visibility != "public-and-private" {
+		cbrURL = fileFallBack(fileMap, c.Visibility, "IBMCLOUD_CONTEXT_BASED_RESTRICTIONS_ENDPOINT", c.Region, cbrURL)
+	}
+	contextBasedRestrictionsClientOptions := &contextbasedrestrictionsv1.Options{
+		Authenticator: authenticator,
+		URL:           envFallBack([]string{"IBMCLOUD_CONTEXT_BASED_RESTRICTIONS_ENDPOINT"}, cbrURL),
+	}
+
+	// Construct the service client.
+	session.contextBasedRestrictionsClient, err = contextbasedrestrictionsv1.NewContextBasedRestrictionsV1(contextBasedRestrictionsClientOptions)
+	if err == nil {
```

Comment:
please check if client not nil condition as well
```
if err == nil && session.contextBasedRestrictionsClient != nil{
 	session.contextBasedRestrictionsClient.Service.EnableRetries(c.RetryCount, c.RetryDelay)
 	session.contextBasedRestrictionsClient.SetDefaultHeaders(gohttp.Header{
			"X-Original-User-Agent": {fmt.Sprintf("terraform-provider-ibm/%s", version.Version)},
		})
}
```

Reply:
checked